pkg/controller/status/status.go

```diff
@@ -778,3 +814,16 @@ func hasPendingCSRUsingCertV1beta1(ctx context.Context, cli client.Client, label
 	}
 	return false, nil
 }
+
+func hasPendingWindowsNodeUpgrade(ctx context.Context, cli client.Client, nodename, expectedVersion string) (bool, error) {
+	node := &corev1.Node{}
+	err := cli.Get(ctx, client.ObjectKey{Name: nodename}, node)
```

Comment:
I'm wondering if we should share the indexer for nodes (that I'm suggesting we add to the controller) with the status so it doesn't need to be querying for node information the controller will already have local access to.